### PR TITLE
GSAGH-518: The GsaGH nightly supports Infinity on unit conversion

### PR DESCRIPTION
**Pull request by SandeepArup**

### CODE REVIEW

File: GsaGH/Parameters/5_Results/4_Subsets/NodeResultsUtility.cs [hunk 1]
```diff
@@ -121,7 +121,7 @@ internal static IQuantity Envelope<T>(this IEnumerable<T> subset, EnvelopeMethod
     internal static Ratio Envelope(this IEnumerable<Ratio?> subset, EnvelopeMethod envelopeType) {
       var val = new Ratio(0, RatioUnit.DecimalFraction);
       for (int i = 0; i < subset.Count(); i++) {
-        if (subset.ElementAt(i) != null) {
+        if (subset.ElementAt(i).HasValue) {
```
Comment (psarras) [inline GsaGH/Parameters/5_Results/4_Subsets/NodeResultsUtility.cs:124]: we don't support Nulls on Ratio?

File: GsaGH/Components/6_Display/ReactionForceDiagrams.cs [hunk 2]
```diff
@@ -143,11 +143,6 @@ protected override void AppendAdditionalComponentMenuItems(ToolStripDropDown men
         momentUnitsMenu,
       };
 
-      if (_lengthUnit == LengthUnit.Undefined) {
-        ToolStripMenuItem modelUnitsMenu = GenerateModelGeometryUnitsMenu("Model geometry");
-        toolStripItems.Insert(0, modelUnitsMenu);
-      }
-
```
Comment (psarras) [inline GsaGH/Components/6_Display/ReactionForceDiagrams.cs:150]: why did we remove those?

File: GsaGH/Parameters/5_Results/5_Extrema/ResultVector6NodeExtremaKeyUtility.cs [hunk 3]
```diff
@@ -148,83 +148,83 @@ private static void UpdateExtrema(IReactionForce item, int nodeId, int permutati
       ref ResultVector6<double?> maxValue, ref ResultVector6<double?> minValue,
       ref ResultVector6<Entity0dExtremaKey> maxKey, ref ResultVector6<Entity0dExtremaKey> minKey) {
 
-      if (item.X != null && ((Force)item.X).Value > maxValue.X) {
-        maxValue.X = ((Force)item.X).Value;
+      if (item.X.HasValue && !double.IsNaN(item.X.Value.Value) && item.X.Value.Value > maxValue.X) {
+        maxValue.X = item.X.Value.Value;
         maxKey.X = new Entity0dExtremaKey(nodeId, permutation);
       }
```
Comment (psarras) [inline GsaGH/Parameters/5_Results/5_Extrema/ResultVector6NodeExtremaKeyUtility.cs:154]: same re nulls here

File: GsaGH/Parameters/5_Results/6_Quantities/SteelUtilisation.cs [hunk 4]
```diff
@@ -5,62 +5,62 @@
...
+    public Ratio? LocalShearU { get; internal set; } = null;
+    public Ratio? LocalShearV { get; internal set; } = null;
+    public Ratio? LocalTorsion { get; internal set; } = null;
+    public Ratio? LocalMajorMoment { get; internal set; } = null;
+    public Ratio? LocalMinorMoment { get; internal set; } = null;
+    public Ratio? MajorBuckling { get; internal set; } = null;
+    public Ratio? MinorBuckling { get; internal set; } = null;
+    public Ratio? LateralTorsionalBuckling { get; internal set; } = null;
+    public Ratio? TorsionalBuckling { get; internal set; } = null;
+    public Ratio? FlexuralBuckling { get; internal set; } = null;
```
Comment (psarras) [inline GsaGH/Parameters/5_Results/6_Quantities/SteelUtilisation.cs:21]: why initialize it?

File: GsaGH/Parameters/5_Results/6_Quantities/SubSpan.cs [hunk 5]
```diff
@@ -5,18 +5,104 @@
...
   public class SubSpan : ISubSpan {
-    public Length StartPosition { get; internal set; }
-    public Length EndPosition { get; internal set; }
-    public Length EffectiveLength { get; internal set; }
-    public Ratio SlendernessRatio { get; internal set; }
+    public Length? StartPosition { get; internal set; } = null;
+    public Length? EndPosition { get; internal set; } = null;
+    public Length? SpanLength { get; internal set; } = null;
+    public Length? EffectiveLength { get; internal set; } = null;
+    public Ratio? SlendernessRatio { get; internal set; } = null;
```
Comment (psarras) [inline GsaGH/Parameters/5_Results/6_Quantities/SubSpan.cs:12]: null?

File: GsaGH/Components/6_Display/ReactionForceDiagrams.cs (hunk 2 above)
Comment (SandeepArup) [reply GsaGH/Components/6_Display/ReactionForceDiagrams.cs:150]: undefined no longer exist. So, Have set default unit to rhinolength unit if active doc not null else meter.  https://github.com/arup-group/GSA-Grasshopper/pull/696/files#diff-6b17d5f6ba38a8a578ae64ec954c7450c81f98364f66ee7229268e5cf7efd5c5

File: GsaGH/Parameters/5_Results/4_Subsets/NodeResultsUtility.cs (hunk 1 above)
Comment (SandeepArup) [reply GsaGH/Parameters/5_Results/4_Subsets/NodeResultsUtility.cs:124]: API return either true vale, NaN or Infinity.

File: GsaGH/Parameters/5_Results/6_Quantities/SubSpan.cs (hunk 5 above)
Comment (SandeepArup) [reply GsaGH/Parameters/5_Results/6_Quantities/SubSpan.cs]: Removed

File: GsaGH/Parameters/5_Results/6_Quantities/SteelUtilisation.cs (hunk 4 above)
Comment (SandeepArup) [reply GsaGH/Parameters/5_Results/6_Quantities/SteelUtilisation.cs]: ReactionForce class was doing initialization in that way and I follwed that. But I think not required and So, have removed

**Pull request merged**